lib/components/ContentModal/ContentModal.js

```diff
@@ -0,0 +1,145 @@
+  MD: 'md',
+  TXT: 'txt',
+  JPEG: 'jpeg',
+  JPG: 'jpg',
+  PNG: 'png',
+  PPTX: 'pptx',
+  XLSX: 'xlsx',
+  DOCX: 'docx',
+  PDF: 'pdf',
+  GENERIC: 'generic',
+  ERROR: 'error',
+};
+
+const getIcon = (alias) =>
+  Object.prototype.hasOwnProperty.call(ICONS, alias.toUpperCase()) &&
+  icons(`./${alias}.svg`).default;
+
+const ContentModal = ({
+  header,
+  titleIcon,
+  title,
+  subtitleIcon,
+  subtitle,
+  children,
+  onRequestClose,
+  onRequestBack,
+}) => {
+  const TitleIcon = getIcon(titleIcon);
+  const SubtitleIcon = getIcon(subtitleIcon);
+
+  return usePortal(
+    <div className={styles.overlay}>
+      <div className={styles.large}>
+        <div className={styles.headerOuter}>
+          <div className={styles.backButton}>
+            <button
+              type="button"
+              className={cn(
+                styles.headerButtonDefault,
+                onRequestBack ? styles.headerButtonEnabled : styles.headerButtonDisabled,
+              )}
+              disabled={!onRequestBack}
+              onClick={onRequestBack}
+            >
+              <Chevron variant={Chevron.VARIANT.THICKER} direction={Chevron.DIRECTION.LEFT} />
+            </button>
+          </div>
+          <div className={styles.headerInner}> {header} </div>
+          <div>
+            <button
+              type="button"
+              className={cn(
+                styles.headerButtonDefault,
+                onRequestClose ? styles.headerButtonEnabled : styles.headerButtonDisabled,
+              )}
+              onClick={onRequestClose}
+            >
+              ✕
+            </button>
+          </div>
+        </div>
+        <div className={styles.body}>
+          <div className={styles.content}>
+            {title && (
+              <div className={styles.title}>
+                {TitleIcon && (
+                  <div className={styles.titleIcon}>
+                    <TitleIcon />
+                  </div>
+                )}
+                {title.key && <div className={styles.titleKey}> {title.key}:&nbsp;</div>}
+                <div className={styles.titleValue}> {title.value} </div>
+              </div>
+            )}
+            {subtitle && (
+              <div
+                className={cn(
+                  styles.subtitleDefault,
+                  SubtitleIcon ? styles.subtitleWithIcon : styles.subtitleWithoutIcon,
+                )}
+              >
+                {SubtitleIcon && (
+                  <div className={styles.subtitleIcon}>
+                    <SubtitleIcon />
+                  </div>
+                )}
+                {subtitle}
+              </div>
+            )}
+            {children}
+          </div>
+        </div>
+      </div>
+    </div>,
+  );
+};
+
+ContentModal.propTypes = {
+  header: PropTypes.node,
+  title: PropTypes.Object,
+  titleIcon: PropTypes.oneOf(Object.values(ICONS)).isRequired,
+  subtitleIcon: PropTypes.oneOf(Object.values(ICONS)).isRequired,
```

Comment:
Are the icons really required? Based on the mocks, the subtitle is at least implied not: https://virtru.invisionapp.com/console/Dashboard-V2-GA-ckbbke6uf17zc01wmwspz6qgt/ckbbkieym18mg01xlzosv2ens/play